TorchSharp/Raw/Blas.tt

```diff
@@ -0,0 +1,188 @@
+﻿<#@ Template Language="C#" #>
+<#@ Assembly Name="System.Core" #>
+<#@ Output Extension=".generated.cs" #>
+<#@ Include File="Types.ttinclude" #>
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace TorchSharp {
+<#
+foreach (var type in TorchTypeDef.Types) {
+
+    string tName = type.ApiName;
+    string tType = type.Storage;
+    string tAccReal = type.AccReal;
+
+    string tTypePtr = "IntPtr"; // "HType";
+#>
+
+    public partial class <#=tName#>Tensor : IDisposable {
+
+        // *** Level 1
+
+        /// <summary>
+        /// <a href="http://www.netlib.org/blas/#_blas_routines">BLAS</a> <c>swap</c> function for <#=tName#> type.
+        /// Swap x and y.
+        /// </summary>
+        /// Corresponds to the following TH definition:
+        /// <code>
+        /// TH_API void THBlas_(swap)(
+        ///     int64_t n, scalar_t *x, int64_t incx, scalar_t *y, int64_t incy);

```

Comment:
This is great! We should use this as standard for the documentation!

Reply:
The above documentation is not correct, and produces a warning, this should be fixed.

I do not think we want to add to the documentation a link to the C API, not sure what value it adds - it is not like the C API has any documentation that could be useful for people anyways.

Our documentation should cover what the API does, so I rather spend the effort on digging through the documentation for how these operations are surfaced in Python and put that documentation there.

Reply:
@migueldeicaza I put those references mostly for the sanity check - if function has a name like `gemm` and 13 parameters, all equally cryptic and with minimal type checks, it is important to make sure that nothing gets lost in translation. Anyway, I have no problem with removing that stuff. Is it OK if I just keep one-line summary for each function? We can add more docs in subsequent PRs

Reply:
We can keep it with regular "//" comments, let us not put it there for the comments that go into the API docs (the "///").

As they currently stand, they also do not compile.

Reply:
@migueldeicaza OK, I've switched to regular `//` comments and removed all XML tags. Both Debug and Release builds worked for me until I've merged in the master. Current master d1bb43fd42e25d222f2eb01f0c327ab1feb7cee1 fails to build the Release, and also has some xmldoc issues:
```
DiskFile.cs(9,18): warning CS1591: Missing XML comment for publicly visible type or member 'DiskFile' [/mnt/c/Users/sergiym/devel/TorchSharp/TorchSharp/TorchSharp.csproj]
THMemoryFile.cs(148,26): warning CS1572: XML comment has a param tag for 'handle', but there is no parameter by that name [/mnt/c/Users/sergiym/devel/TorchSharp/TorchSharp/TorchSharp.csproj]
THMemoryFile.cs(150,39): warning CS1573: Parameter 'storage' has no matching param tag in the XML comment for 'MemoryFile.MemoryFile(MemoryFile.CharStorage, string)' (but other parameters do) [/mnt/c/Users/sergiym/devel/TorchSharp/TorchSharp/TorchSharp.csproj]
/usr/share/dotnet/sdk/2.1.403/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(198,5): error NETSDK1004: Assets file '/mnt/c/Users/sergiym/devel/TorchSharp/TorchSharp/C:/Users/sergiym/devel/TorchSharp/TorchSharp/obj/project.assets.json' not found. Run a NuGet package restore to generate this file. [/mnt/c/Users/sergiym/devel/TorchSharp/TorchSharp/TorchSharp.csproj]
    3 Warning(s)
    1 Error(s)
```
so it probably isn't related to this PR